### PR TITLE
feat: create ConfirmInfoRowUrl

### LINKED ISSUE #21286: Create component to display origin information
We need a component that should display the origin for the confirmation:

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWtkQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--53c30e27f65806b7b6f18ece82e1bc12be9e9099/Screenshot%202023-10-10%20at%204.15.26%E2%80%AFPM.png)

Please note that not every request will have a contract.

Designs: https://www.figma.com/file/uACzbBjwU2tV30UuewYiTx/Confirmation-redesign-V2?type=design&node-id=775-28145&mode=design&t=fcwGIVZS3lkt9rOO-0

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

Create the reusable url row component. This will be used to display the origin and other URLs. 
- displays URL that will text wrap
- if the string is a valid input for javascript URL method, it will strip away the protocol from the text, else, displays the string as is
- displays warning "HTTP" tag + icon when URL is HTTP instead of HTTPS
  - we have a lazy way to extract HTTP now which only checks from the provided URL string. in the future we may bake in logic infer the protocol e.g. infer protocol based on port number. 
  **see:** https://github.com/MetaMask/core/blob/27047d2381b1995a6da32971925233a786f74a1f/packages/controller-utils/src/siwe.ts

Note:
- SIWE supports domain binding which checks the specified signature domain against the origin domain. If the domain in invalid, we will need to show an error state. This is not supported in this PR as we will be supporting SIWE (variation of personal_sign transaction) following the plain personal_sign page support

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21286

## **Manual testing steps**

1. yarn storybook
2. http://localhost:6006/?path=/docs/components-app-confirm-inforowurl--docs

## **Screenshots/Recordings**

Screenshot 2023-12-18 at 6 07 06 PM

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/confirm/info/row/url.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  IconColor,
+  TextColor,
+  TextVariant,
+  BackgroundColor,
+} from '../../../../../helpers/constants/design-system';
+
+export type ConfirmInfoRowUrlProps = {
+  url: string;
+};
+
```
Comment (jpuri) [inline ui/components/app/confirm/info/row/url.tsx:23]: The name `ConfirmInfoRowUrl` is un-intuitive. We should not couple these components to `ConfirmInfoRow`. These can be independent component which can be wrapped in `ConfirmInfoRow`.

File: ui/components/app/confirm/info/row/url.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  } catch (e) {
+    console.log(`ConfirmInfoRowUrl: Error parsing url: ${urlString}`);
+  }
+
+  const isHTTP = url?.protocol === 'http:';
+  const isValidUrl = Boolean(url);
+
+  const displayUrl = isValidUrl
+    ? `${url?.host}${url?.pathname === '/' ? '' : url?.pathname}`
+    : urlString;
```
Comment (jpuri) [inline ui/components/app/confirm/info/row/url.tsx:40]: I think we should not parse URL as above to get display string, we should display the url string as it is.

**Comment by jpuri** [comment]
Code changes look good, I left some small feedbacks.

File: ui/components/app/confirm/info/row/url.tsx (hunk 2 above)
Comment (digiwand) [reply ui/components/app/confirm/info/row/url.tsx]: sounds good to me. updated [df5ba34](https://github.com/MetaMask/metamask-extension/pull/22293/commits/df5ba34cb1e2671754107d4dd9a3e2ec126056bc)

**Review by jpuri — APPROVED** [review]

File: ui/components/app/confirm/info/row/url.tsx (hunk 1 above)
Comment (digiwand) [reply ui/components/app/confirm/info/row/url.tsx:23]: note: if we change this, we can do this in a future PR and apply changes to all relevant components

**Review by segun — APPROVED** [review]

**Pull request merged**